man/myisam_ftdump.1
'\" t
.\"     Title: myisam_ftdump
.\"    Author: [FIXME: author] [see http://docbook.sf.net/el/author]
.\" Generator: DocBook XSL Stylesheets v1.79.1 <http://docbook.sf.net/>
.\"      Date: 06/25/2026
.\"    Manual: MySQL Database System
.\"    Source: MySQL 9.7
.\"  Language: English
.\"
.TH "MYISAM_FTDUMP" "1" "06/25/2026" "MySQL 9\&.7" "MySQL Database System"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
myisam_ftdump \- display full\-text index information
.SH "SYNOPSIS"
.HP \w'\fBmyisam_ftdump\ [\fR\fB\fIoptions\fR\fR\fB]\ \fR\fB\fItbl_name\fR\fR\fB\ \fR\fB\fIindex_num\fR\fR\ 'u
\fBmyisam_ftdump [\fR\fB\fIoptions\fR\fR\fB] \fR\fB\fItbl_name\fR\fR\fB \fR\fB\fIindex_num\fR\fR
.SH "DESCRIPTION"
.PP
\fBmyisam_ftdump\fR
displays information about
FULLTEXT
indexes in
MyISAM
tables\&. It reads the
MyISAM
index file directly, so it must be run on the server host where the table is located\&. Before using
\fBmyisam_ftdump\fR, be sure to issue a
FLUSH TABLES
statement first if the server is running\&.
.PP
\fBmyisam_ftdump\fR
scans and dumps the entire index, which is not particularly fast\&. On the other hand, the distribution of words changes infrequently, so it need not be run often\&.
.PP
Invoke
\fBmyisam_ftdump\fR
like this:
.sp
.if n \{\
.RS 4
.\}
.nf
myisam_ftdump [\fIoptions\fR] \fItbl_name\fR \fIindex_num\fR
.fi
.if n \{\
.RE
.\}
.PP
The
\fItbl_name\fR
argument should be the name of a
MyISAM
table\&. You can also specify a table by naming its index file (the file with the
\&.MYI
suffix)\&. If you do not invoke
\fBmyisam_ftdump\fR
in the directory where the table files are located, the table or index file name must be preceded by the path name to the table\*(Aqs database directory\&. Index numbers begin with 0\&.
.PP
Example: Suppose that the
test
database contains a table named
mytexttable
that has the following definition:
.sp
.if n \{\
.RS 4
.\}
.nf
CREATE TABLE mytexttable
(
  id   INT NOT NULL,
  txt  TEXT NOT NULL,
  PRIMARY KEY (id),
  FULLTEXT (txt)
) ENGINE=MyISAM;
.fi
.if n \{\
.RE
.\}
.PP
The index on
id
is index 0 and the
FULLTEXT
index on
txt
is index 1\&. If your working directory is the
test
database directory, invoke
\fBmyisam_ftdump\fR
as follows:
.sp
.if n \{\
.RS 4
.\}
.nf
myisam_ftdump mytexttable 1
.fi
.if n \{\
.RE
.\}
.PP
If the path name to the
test
database directory is
/usr/local/mysql/data/test, you can also specify the table name argument using that path name\&. This is useful if you do not invoke
\fBmyisam_ftdump\fR
in the database directory:
.sp
.if n \{\
.RS 4
.\}
.nf
myisam_ftdump /usr/local/mysql/data/test/mytexttable 1
.fi
.if n \{\
.RE
.\}
.PP
You can use
\fBmyisam_ftdump\fR
to generate a list of index entries in order of frequency of occurrence like this on Unix\-like systems:
.sp
.if n \{\
.RS 4
.\}
.nf
myisam_ftdump \-c mytexttable 1 | sort \-r
.fi
.if n \{\
.RE
.\}
.PP
On Windows, use:
.sp
.if n \{\
.RS 4
.\}
.nf
myisam_ftdump \-c mytexttable 1 | sort /R
.fi
.if n \{\
.RE
.\}
.PP
\fBmyisam_ftdump\fR
supports the following options:
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-help\fR,
\fB\-h\fR
\fB\-?\fR
.TS
allbox tab(:);
lB l.
T{
Command-Line Format
T}:T{
--help
T}
.TE
.sp 1
Display a help message and exit\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-count\fR,
\fB\-c\fR
.TS
allbox tab(:);
lB l.
T{
Command-Line Format
T}:T{
--count
T}
.TE
.sp 1
Calculate per\-word statistics (counts and global weights)\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-dump\fR,
\fB\-d\fR
.TS
allbox tab(:);
lB l.
T{
Command-Line Format
T}:T{
--dump
T}
.TE
.sp 1
Dump the index, including data offsets and word weights\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-length\fR,
\fB\-l\fR
.TS
allbox tab(:);
lB l.
T{
Command-Line Format
T}:T{
--length
T}
.TE
.sp 1
Report the length distribution\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-stats\fR,
\fB\-s\fR
.TS
allbox tab(:);
lB l.
T{
Command-Line Format
T}:T{
--stats
T}
.TE
.sp 1
Report global index statistics\&. This is the default operation if no other operation is specified\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-verbose\fR,
\fB\-v\fR
.TS
allbox tab(:);
lB l.
T{
Command-Line Format
T}:T{
--verbose
T}
.TE
.sp 1
Verbose mode\&. Print more output about what the program does\&.
.RE
.SH "COPYRIGHT"
.br
.PP
Copyright \(co 1997, 2026, Oracle and/or its affiliates.
.PP
This documentation is free software; you can redistribute it and/or modify it only under the terms of the GNU General Public License as published by the Free Software Foundation; version 2 of the License.
.PP
This documentation is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
.PP
You should have received a copy of the GNU General Public License along with the program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA or see http://www.gnu.org/licenses/.
.sp
.SH "SEE ALSO"
For more information, please refer to the MySQL Reference Manual,
which may already be installed locally and which is also available
online at http://dev.mysql.com/doc/.
.SH AUTHOR
Oracle Corporation (http://dev.mysql.com/).


ChangeLog
2010-03-18	The InnoDB Team

	* CMakeLists.txt:
	Fix Bug#52102 InnoDB Plugin shows performance drop compared to
	InnoDB (Windows)

2010-03-18	The InnoDB Team

	* buf0buf.ic:
	When comparing the time of the first access to a block against
	innodb_old_blocks_time, use 32-bit arithmetics. The comparison was
	incorrect on 64-bit systems.

2010-03-11	The InnoDB Team

	* buf0buf.h, buf0buf.ic:
	Fix and clarify the latching of some buf_block_t members.
	Note that check_index_page_at_flush is not protected by any mutex.
	Note and assert that lock_hash_val is protected by the rw-latch.

2010-03-10	The InnoDB Team

	* trx/trx0sys.c:
	Fix Bug#51653 outdated reference to set-variable

2010-03-10	The InnoDB Team

	* handler/ha_innodb.cc, mysql-test/innodb_bug21704.result,
	mysql-test/innodb_bug47621.result, mysql-test/innodb_bug47621.test:
	Fix Bug#47621 MySQL and InnoDB data dictionaries will become out of
	sync when renaming columns

2010-03-10	The InnoDB Team

	* handler/ha_innodb.cc:
	Fix Bug#51356 Many Valgrind errors in error messages
	with concurrent DDL

2010-03-10	The InnoDB Team

	* handler/ha_innodb.cc, handler/handler0alter.cc,
	mysql-test/innodb_bug51378.result, mysql-test/innodb_bug51378.test:
	Fix Bug#51378 Init 'ref_length' to correct value, in case an out
	of bound MySQL primary_key

2010-03-10	The InnoDB Team

	* log/log0recv.c:
	Remove a bogus assertion about page numbers exceeding 0x90000000
	in the redo log. Abort when encountering a corrupted redo log
	record, unless innodb_force_recovery is set.

2010-03-09	The InnoDB Team

	* handler/ha_innodb.cc:
	Make SHOW ENGINE INNODB MUTEX STATUS display SUM(os_waits)
	for the buffer pool block mutexes and locks.

2010-03-08	The InnoDB Team

	* fil/fil0fil.c:
	Fix ALTER TABLE ... IMPORT TABLESPACE of compressed tables.

2010-03-03	The InnoDB Team

	* handler/handler0alter.cc, innodb-index.result, innodb-index.test,
	innodb.result, innodb.test:
	Disallow a duplicate index name when creating an index.

2010-02-11	The InnoDB Team

	* include/mem0mem.h, include/mem0mem.ic, mem/mem0mem.c:
	Fix Bug#49535 Available memory check slows down crash
	recovery tens of times

2010-02-09	The InnoDB Team

	* buf/buf0buf.c:
	Fix Bug#38901 InnoDB logs error repeatedly when trying to load
	page into buffer pool

2010-02-09	The InnoDB Team

	* srv/srv0srv.c:
	Let the master thread sleep if the amount of work to be done is
	calibrated as taking less than a second.

2010-02-04	The InnoDB Team

	* btr/btr0btr.c, btr/btr0cur.c, btr/btr0pcur.c, buf/buf0buf.c,
	include/btr0btr.h, include/btr0cur.h, include/btr0pcur.h,
	include/btr0pcur.ic, include/buf0buf.h, row/row0ins.c, row/row0sel.c:
	Pass the file name and line number of the caller of the
	b-tree cursor functions to the buffer pool requests, in order
	to make the latch diagnostics more accurate.

2010-02-03	The InnoDB Team

	* lock/lock0lock.c:
	Fix Bug#49001 SHOW INNODB STATUS deadlock info incorrect
	when deadlock detection aborts

2010-02-03	The InnoDB Team

	* buf/buf0lru.c:
	Fix Bug#35077 Very slow DROP TABLE (ALTER TABLE, OPTIMIZE TABLE)
	on compressed tables

2010-02-03	The InnoDB Team

	* handler/ha_innodb.cc, include/row0mysql.h, row/row0mysql.c:
	Clean up CHECK TABLE error handling.

2010-02-01	The InnoDB Team

	* handler/ha_innodb.cc, mysql-test/innodb-autoinc.test,
	mysql-test/innodb-autoinc.result,
	mysql-test/innodb-autoinc-44030.test,
	mysql-test/innodb-autoinc-44030.result:
	Fix Bug#49497 Error 1467 (ER_AUTOINC_READ_FAILED) on inserting
	a negative value

2010-01-27	The InnoDB Team

	* include/row0mysql.h, log/log0recv.c, row/row0mysql.c:
	Drop temporary tables at startup.
	This addresses the third aspect of
	Bug#41609 Crash recovery does not work for InnoDB temporary tables.

2010-01-21	The InnoDB Team

	* buf/buf0buf.c:
	Do not merge buffered inserts to compressed pages before
	the redo log has been applied in crash recovery.

2010-01-13	The InnoDB Team

	* row/row0sel.c:
	On the READ UNCOMMITTED isolation level, do not attempt to access
	a clustered index record that has been marked for deletion. The
	built-in InnoDB in MySQL 5.1 and earlier would attempt to retrieve
	a previous version of the record in this case.

2010-01-13	The InnoDB Team

	* buf/buf0buf.c:
	When disabling the adaptive hash index, check the block state
	before checking block->is_hashed, because the latter may be
	uninitialized right after server startup.

2010-01-12	The InnoDB Team

	* handler/ha_innodb.cc, handler/ha_innodb.h:
	Fix Bug#46193 crash when accessing tables after enabling
	innodb_force_recovery option

2010-01-12	The InnoDB Team

	* row/row0mysql.c:
	Fix Bug#49238 Creating/Dropping a temporary table while at 1023
	transactions will cause assert.

2009-12-02	The InnoDB Team

	* srv/srv0start.c:
	Display the zlib version number at startup.
	InnoDB compressed tables use zlib, and the implementation depends
	on the zlib function compressBound(), whose definition was slightly
	changed in zlib version 1.2.3.1 in 2006. MySQL bundles zlib 1.2.3
	from 2005, but some installations use a more recent zlib.

2009-11-30	The InnoDB Team

	* dict/dict0crea.c, dict/dict0mem.c, dict/dict0load.c,
	dict/dict0boot.c, fil/fil0fil.c, handler/ha_innodb.cc,
	include/dict0mem.h, row/row0mysql.c:
	Fix the bogus warning messages for non-existing temporary
	tables that were reported in
	Bug#41609 Crash recovery does not work for InnoDB temporary tables.
	The actual crash recovery bug was corrected on 2009-04-29.

2009-11-27	The InnoDB Team

	InnoDB Plugin 1.0.6 released

2009-11-20	The InnoDB Team

	* handler/ha_innodb.cc:
	Add a workaround to prevent a crash due to Bug#45961 DDL on
	partitioned innodb tables leaves data dictionary in an inconsistent
	state

2009-11-19	The InnoDB Team

	* btr/btr0btr.c:
	Fix Bug#48469 when innodb tablespace is configured too small, crash
	and corruption!

2009-11-19	The InnoDB Team

	* data/data0type.c:
	Fix Bug#48526 Data type for float and double is incorrectly reported
	in InnoDB table monitor


storage/myisam/myisam_ftdump.cc
/* Copyright (c) 2001, 2026, Oracle and/or its affiliates.

   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License, version 2.0,
   as published by the Free Software Foundation.

   This program is designed to work with certain software (including
   but not limited to OpenSSL) that is licensed under separate terms,
   as designated in a particular file or component or in included license
   documentation.  The authors of MySQL hereby grant you an additional
   permission to link the program and your derivative works with the
   separately licensed software that they have either included with
   the program or referenced in the documentation.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License, version 2.0, for more details.

   You should have received a copy of the GNU General Public License
   along with this program; if not, write to the Free Software
   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */

/* Written by Sergei A. Golubchik, who has a shared copyright to this code
   added support for long options (my_getopt) 22.5.2002 by Jani Tolonen */

#include <fcntl.h>
#include <sys/types.h>
#include <cstdlib>

#include "m_string.h"
#include "my_compiler.h"
#include "my_getopt.h"
#include "my_inttypes.h"
#include "mysql/strings/m_ctype.h"
#include "print_version.h"
#include "storage/myisam/ftdefs.h"
#include "storage/myisam/myisamdef.h"
#include "welcome_copyright_notice.h"

[[noreturn]] static void usage();
static void complain(int val);
static bool get_one_option(int, const struct my_option *, char *);

static int count = 0, stats = 0, dump = 0, lstats = 0;
static bool verbose;
static char *query = nullptr;
static uint lengths[256];

#define MAX_LEN (HA_FT_MAXBYTELEN + 10)
#define HOW_OFTEN_TO_WRITE 10000

static struct my_option my_long_options[] = {
    {"help", 'h', "Display help and exit.", nullptr, nullptr, nullptr,
     GET_NO_ARG, NO_ARG, 0, 0, 0, nullptr, 0, nullptr},
    {"help", '?', "Synonym for -h.", nullptr, nullptr, nullptr, GET_NO_ARG,
     NO_ARG, 0, 0, 0, nullptr, 0, nullptr},
    {"count", 'c', "Calculate per-word stats (counts and global weights).",
     nullptr, nullptr, nullptr, GET_NO_ARG, NO_ARG, 0, 0, 0, nullptr, 0,
     nullptr},
    {"dump", 'd', "Dump index (incl. data offsets and word weights).", nullptr,
     nullptr, nullptr, GET_NO_ARG, NO_ARG, 0, 0, 0, nullptr, 0, nullptr},
    {"length", 'l', "Report length distribution.", nullptr, nullptr, nullptr,
     GET_NO_ARG, NO_ARG, 0, 0, 0, nullptr, 0, nullptr},
    {"stats", 's', "Report global stats.", nullptr, nullptr, nullptr,
     GET_NO_ARG, NO_ARG, 0, 0, 0, nullptr, 0, nullptr},
    {"verbose", 'v', "Be verbose.", &verbose, &verbose, nullptr, GET_BOOL,
     NO_ARG, 0, 0, 0, nullptr, 0, nullptr},
    {nullptr, 0, nullptr, nullptr, nullptr, nullptr, GET_NO_ARG, NO_ARG, 0, 0,
     0, nullptr, 0, nullptr}};

extern st_keycache_thread_var *keycache_thread_var() {
  return &main_thread_keycache_var;
}

int main(int argc, char *argv[]) {
  int error = 0, subkeys;
  uint keylen, keylen2 = 0, inx, doc_cnt = 0;
  float weight = 1.0;
  double gws, min_gws = 0, avg_gws = 0;
  MI_INFO *info;
  char buf[MAX_LEN], buf2[MAX_LEN], buf_maxlen[MAX_LEN], buf_min_gws[MAX_LEN];
  ulong total = 0, maxlen = 0, uniq = 0, max_doc_cnt = 0;
  struct {
    MI_INFO *info;
  } aio0, *aio = &aio0; /* for GWS_IN_USE */

  MY_INIT(argv[0]);

  memset(&main_thread_keycache_var, 0, sizeof(st_keycache_thread_var));
  mysql_cond_init(PSI_NOT_INSTRUMENTED, &main_thread_keycache_var.suspend);

  if ((error = handle_options(&argc, &argv, my_long_options, get_one_option)))
    exit(error);
  if (count || dump) verbose = false;
  if (!count && !dump && !lstats && !query) stats = 1;

  if (verbose) setbuf(stdout, nullptr);

  if (argc < 2) usage();

  {
    char *end;
    inx = (uint)my_strtoll(argv[1], &end, 10);
    if (*end) usage();
  }

  init_key_cache(dflt_key_cache, MI_KEY_BLOCK_LENGTH, USE_BUFFER_INIT, 0, 0);

  if (!(info = mi_open(argv[0], O_RDONLY,
                       HA_OPEN_ABORT_IF_LOCKED | HA_OPEN_FROM_SQL_LAYER))) {
    error = my_errno();
    goto err;
  }

  *buf2 = 0;
  aio->info = info;

  if ((inx >= info->s->base.keys) ||
      !(info->s->keyinfo[inx].flag & HA_FULLTEXT)) {
    printf("Key %d in table %s is not a FULLTEXT key\n", inx, info->filename);
    goto err;
  }

  mi_lock_database(info, F_EXTRA_LCK);

  info->lastpos = HA_OFFSET_ERROR;
  info->update |= HA_STATE_PREV_FOUND;

  while (!(error = mi_rnext(info, nullptr, inx))) {
    keylen = *(info->lastkey);

    subkeys = ft_sintXkorr(info->lastkey + keylen + 1);
    if (subkeys >= 0) weight = ft_floatXget(info->lastkey + keylen + 1);

    snprintf(buf, MAX_LEN, "%.*s", (int)keylen, info->lastkey + 1);
    my_casedn_str(default_charset_info, buf);
    total++;
    lengths[keylen]++;

    if (count || stats) {
      if (strcmp(buf, buf2) != 0) {
        if (*buf2) {
          uniq++;
          avg_gws += gws = GWS_IN_USE;
          if (count) printf("%9u %20.7f %s\n", doc_cnt, gws, buf2);
          if (maxlen < keylen2) {
            maxlen = keylen2;
            my_stpcpy(buf_maxlen, buf2);
          }
          if (max_doc_cnt < doc_cnt) {
            max_doc_cnt = doc_cnt;
            my_stpcpy(buf_min_gws, buf2);
            min_gws = gws;
          }
        }
        my_stpcpy(buf2, buf);
        keylen2 = keylen;
        doc_cnt = 0;
      }
      doc_cnt += (subkeys >= 0 ? 1 : -subkeys);
    }
    if (dump) {
      if (subkeys >= 0)
        printf("%9lx %20.7f %s\n", (long)info->lastpos, weight, buf);
      else
        printf("%9lx => %17d %s\n", (long)info->lastpos, -subkeys, buf);
    }
    if (verbose && (total % HOW_OFTEN_TO_WRITE) == 0) printf("%10ld\r", total);
  }
  mi_lock_database(info, F_UNLCK);

  if (count || stats) {
    if (*buf2) {
      uniq++;
      avg_gws += gws = GWS_IN_USE;
      if (count) printf("%9u %20.7f %s\n", doc_cnt, gws, buf2);
      if (maxlen < keylen2) {
        maxlen = keylen2;
        my_stpcpy(buf_maxlen, buf2);
      }
      if (max_doc_cnt < doc_cnt) {
        max_doc_cnt = doc_cnt;
        my_stpcpy(buf_min_gws, buf2);
        min_gws = gws;
      }
    }
  }

  if (stats) {
    count = 0;
    for (inx = 0; inx < 256; inx++) {
      count += lengths[inx];
      if ((ulong)count >= total / 2) break;
    }
    printf(
        "Total rows: %lu\nTotal words: %lu\n"
        "Unique words: %lu\nLongest word: %lu chars (%s)\n"
        "Median length: %u\n"
        "Average global weight: %f\n"
        "Most common word: %lu times, weight: %f (%s)\n",
        (long)info->state->records, total, uniq, maxlen, buf_maxlen, inx,
        avg_gws / uniq, max_doc_cnt, min_gws, buf_min_gws);
  }
  if (lstats) {
    count = 0;
    for (inx = 0; inx < 256; inx++) {
      count += lengths[inx];
      if (count && lengths[inx])
        printf("%3u: %10lu %5.2f%% %20lu %4.1f%%\n", inx, (ulong)lengths[inx],
               100.0 * lengths[inx] / total, (ulong)count,
               100.0 * count / total);
    }
  }

err:
  if (error && error != HA_ERR_END_OF_FILE)
    printf("got error %d\n", my_errno());
  if (info) mi_close(info);
  mysql_cond_destroy(&main_thread_keycache_var.suspend);
  return 0;
}

static bool get_one_option(int optid,
                           const struct my_option *opt [[maybe_unused]],
                           char *argument [[maybe_unused]]) {
  switch (optid) {
    case 'd':
      dump = 1;
      complain(count || query);
      break;
    case 's':
      stats = 1;
      complain(query != nullptr);
      break;
    case 'c':
      count = 1;
      complain(dump || query);
      break;
    case 'l':
      lstats = 1;
      complain(query != nullptr);
      break;
    case '?':
    case 'h':
      usage();
  }
  return false;
}

static void usage() {
  print_version();
  puts(ORACLE_WELCOME_COPYRIGHT_NOTICE("2002"));

  printf("Use: myisam_ftdump <table_name> <index_num>\n");
  my_print_help(my_long_options);
  my_print_variables(my_long_options);
  exit(1);
}

static void complain(int val) /* Kinda assert :-)  */
{
  if (val) {
    printf("You cannot use these options together!\n");
    exit(1);
  }
}

#include "storage/myisam/mi_extrafunc.h"
